packages/core/core/src/utils/nuxt/_proxyUtils.ts

```diff
@@ -1,13 +1,35 @@
 import { Context as NuxtContext } from '@nuxt/types';
 import merge from 'lodash-es/merge';
 import { ApiClientMethod } from './../../types';
+import { AxiosInstance } from 'axios';
 
 interface CreateProxiedApiParams {
   givenApi: Record<string, ApiClientMethod>;
   client: any;
   tag: string;
 }
 
+const getUrl = (nxtContent: NuxtContext, endpoint: string) => {
+  const { base, req } = nxtContent;
+  const isServer = process.server;
+  const isClient = process.client;
+  const hasApiTag = /\/api\//gi.test(endpoint);
+  const apiEndpoint = hasApiTag ? endpoint : `api/${endpoint}`;
+
+  if (!req) {
+    return isClient && !isServer
+      ? `${window.location.origin}/${apiEndpoint}`
+      : apiEndpoint;
+  }
+
+  const { headers } = req;
+  const isHttps = require('is-https')(req);
+  const scheme = isHttps ? 'https' : 'http';
+  const host = headers['x-forwarded-host'] || headers.host;
```

Comment:
The host headers are being fetched from the server request itself. There is no way of interfering with this.

Reply:
I will need help with those tests, because we need to find a way to fix this without changing the base structure, or we need to release a `minor` version of the lib.

Reply:
I've sent you instruction regarding testing